### DISCUSSION TITLE
Starting with Eclipse GLSP

### DISCUSSION THREAD

**Discussion by JValGLZ** [eclipse-glsp/glsp#1222] [General] [answered]
I am an undergraduate student looking to use this framework for a school project.  I have experience in Java and am learning typescript so I want to work with the Java-EMF-Theia Template. I am trying to get the backend of an old Java modeling software tool and move it so that it is part of this innovative graphical framework. As I have been looking at the code, I understand that there are concepts that I must become familiar with, and wanted to know if there was a guide for new users to the framework. I have tried looking online, but I do not see tutorials explaining examples of how one would go about starting their diagram or what concepts to study other than programming languages. I would like to start with learning the steps to connect the client and server by adding a completely new visual entity. I would appreciate it if it could be explained. Thank you.

**Comment by planger** [comment, marked as answer]
Hey there!

It's great to hear about your interest in using the Eclipse GLSP for your school project.

For an introduction, the [Getting Started Guide](https://eclipse.dev/glsp/documentation/gettingstarted/) on the Eclipse GLSP documentation is the perfect place to start. It'll guide you through the initial steps and fundamental concepts required to get your application running, based on the  [Java-EMF-Theia Template](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/java-emf-theia) in your case.

The Java-EMF-Theia template you mentioned is a solid starting point for integrating the backend of an older Java modeling tool. Next, you'll probably may want to look at integrating your [custom source model](https://eclipse.dev/glsp/documentation/sourcemodel/) from the old modeling tool. I'm not sure whether this is EMF-based or not, but essentially you'll just need to load the source model and [transform it into the graphical model](https://eclipse.dev/glsp/documentation/gmodel/). Then you can start working on the [rendering](https://eclipse.dev/glsp/documentation/rendering/) and implement the [edit operations](https://eclipse.dev/glsp/documentation/modeloperations/). I've linked the respective parts of the GLSP documentation above.

Additionally, there's a tutorial recording available [here](https://www.youtube.com/watch?v=f5gVKh04Oho) which, despite being slightly outdated in some aspects, provides a solid overview of the framework and can be quite helpful in visualizing how everything comes together.

Diving into a new framework can be challenging, but with these resources, I hope you'll be well on your way. Don't hesitate to reach out to the community if you have more questions or need further assistance.

Best of luck with your project!

**Reply by JValGLZ** [reply]
Thank you for your help!
